morf-testsupport/src/main/java/org/alfasoftware/morf/jdbc/AbstractSqlDialectTest.java

```diff
@@ -3360,17 +3361,35 @@ public void testHints() {
         .withCustomHint(() -> "CustomHint")
       )
     );
+
+    assertEquals(
+      expectedHints8a(),
+      testDialect.convertStatementToSQL(
+        select()
+        .from(new TableReference("SCHEMA2", "Foo"))
+        .withDialectSpecificHint(provideDatabaseType(), "index(customer cust_primary_key_idx)")

```

Comment:
```
  .withDialectSpecificHint(provideDatabaseType(), "index(customer cust_primary_key_idx)")
  .withDialectSpecificHint("SOMETHING_ELSE", "unused_hint()")
```
That way you also make sure dialects do not react to hints that do not belong to them :)

Reply:
Done